internal/cmd/workspaces.go

```diff
@@ -85,15 +86,25 @@ func lsWorkspacesCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			workspaces, err := getWorkspaces(ctx, client, user)
-			if err != nil {
-				return err
-			}
-			if provider != "" {
+			var workspaces []coder.Workspace
+			if all {
+				var err error
```

Comment:
You shouldn't need the `var err`s here or in the other if branches since it's already defined above, then you can condense the `if err` check to be underneath the if branches, so each branch is just the `workspaces, err = getX()` statement.

edit: check my suggestion